agent/provider/src/rules.rs

```diff
@@ -187,16 +187,29 @@ impl RulesManager {
 
     pub fn spawn_file_monitors(&self) -> Result<(FileMonitor, FileMonitor, FileMonitor)> {
         let rulestore_monitor = {
+            let cert_dir = self.cert_dir.clone();
             let manager = self.clone();
-            let handler = move |p: PathBuf| match manager.rulestore.reload() {
-                Ok(()) => {
-                    log::info!("rulestore updated from {}", p.display());
+            let handler = move |p: PathBuf| {
+                // Reload also keystore to avoid file-monitor race when doing `import-cert`
+                match manager.keystore.reload(&cert_dir) {
+                    Ok(()) => {
+                        log::info!("Trusted keystore updated because rulestore changed");
```

Comment:
Will it not be printed every time yagna starts up?

Reply:
No, it will be printed only when rulestore changes (either from CLI or from manual changes to file directly)